pkg/datagatherer/k8s/generic_test.go

```diff
@@ -86,6 +86,9 @@ func TestGenericGatherer_Fetch(t *testing.T) {
 				getObject("foobar/v1", "Foo", "testfoo", "nottestns"),
 			),
 		},
+		// Note that we can't test use of fieldSelector to exclude namespaces
+		// here as the as the fake client does not implement it.
+		// See go/pkg/mod/k8s.io/client-go@v0.17.0/dynamic/fake/simple.go:291
```

Comment:
I think it _is_ possibly to define a custom reactor however, that should be able to access request parameters like this in order to ensure that the expected selector is actually being sent to the server when this request is made.

Alternatively/additionally, we could look at a light-weight integration test for this once there's a few more places that need it. Perhaps worth adding a TODO for the integration test, and opening an issue to track adding an integration test framework?

Reply:
Yeah I've spent a bit of time looking into this but I think we're best to open a separate issue for it.